### PR TITLE
Update OWNERS to only require project mentors' approval

### PR DESCRIPTION
**Pull request by rzetelskik**
**Description of your changes:** This PR updates the OWNERS file to only require approval from the project mentors.

**Which issue is resolved by this Pull Request:**
Resolves #

/kind cleanup
/priority important-longterm

### CODE REVIEW

File: OWNERS [hunk 1]
```diff
@@ -3,14 +3,9 @@
...
+    - rzetelskik
     reviewers:
-    - tnozicka
     - zimnx
     - rzetelskik
-  "go\\.(mod|sum)$":
-    required_reviewers:
-    - tnozicka
-    labels:
-    - area/dependency
```
Comment (tnozicka) [inline OWNERS:16]: technically this could stay

**Review by tnozicka — APPROVED** [review]
/approve
(lgtm)

**Review by zimnx — APPROVED** [review]
/lgtm

**Pull request merged**